### ISSUE TITLE
GcmJobSchedulerService.createSchedulerFor tries to expose internal class GcmScheduler

### ISSUE THREAD

**Issue by consp1racy** [yigit/android-priority-jobqueue#237]
GcmScheduler is package private class and cannot be used in configuration builder. Same goes for FrameworkScheduler.

**Proposed solution:**

Change contract from 

    public static GcmScheduler createSchedulerFor(Context appContext, Class<? extends GcmJobSchedulerService> klass)

to

    public static Scheduler createSchedulerFor(Context appContext, Class<? extends 

Repeat for FrameworkJobSchedulerService.

**Workaround (Kotlin):**

```
object com.birbit.android.jobqueue.scheduling.SchedulerFactory {
    fun createSchedulerForGcmJobSchdulerService(appContext: Context, klass: Class<out GcmJobSchedulerService>): Scheduler {
        return GcmJobSchedulerService.createSchedulerFor(appContext, klass)
    }

    fun createSchedulerForFrameworkJobSchedulerService(appContext: Context, klass: Class<out FrameworkJobSchedulerService>): Scheduler {
        return FrameworkJobSchedulerService.createSchedulerFor(appContext, klass)
    }
}
```

**Comment by yigit** [comment]
You are not expected to use it directly.
I'm not sure what the problem is, did you check the configuration in the sample app?

https://github.com/yigit/android-priority-jobqueue/blob/master/examples/twitter/TwitterClient/src/com/birbit/android/jobqueue/examples/twitter/TwitterApplication.java

**Comment by consp1racy** [comment]
Understood. This may or may not be just Kotlin related. I basically copied the sample code and converted to Kotlin. Problem is Kotlin infers the return value but the inferred class is package local and marked as an error.

![image](https://cloud.githubusercontent.com/assets/4589773/17883288/00df58dc-6912-11e6-915f-a9f88d05eba6.png)

**Comment by yigit** [comment]
I guess kotlin IJ plugin is going too aggressive. Does the code compile?
I can change the return type, i don't think it will break anybody but since it will be an API change, it would need to wait for 2.1. I can also make the class public but it will add new  API so not preferred.

**Comment by consp1racy** [comment]
Nope, original code does not compile. Since nobody else complained, scheduling an API change for 2.1 should be fine, what do you think?

![image](https://cloud.githubusercontent.com/assets/4589773/17981473/2868479e-6b04-11e6-8dd8-428cedea5c55.png)

**Comment by yigit** [comment]
interesting. Sample app still compiles so there is no problem there but seems like kotlin does not like it. I'll make the api change.

**Comment by otomatik** [comment]
@consp1racy I encounter the same issue, where do define the ```com.birbit.android.jobqueue.scheduling.SchedulerFactory```object in your project to make the workaround?
thanks in advance

**Comment by consp1racy** [comment]
@otomatik You just answered your own question. ;)

You declare a `SchedulerFactory` object in `com.birbit.android.jobqueue.scheduling` package.
